### PR TITLE
Add docsearch

### PR DESCRIPTION
**Pull request by SMillerDev**
This PR implements https://community.algolia.com/docsearch/ to make it easier for brew.sh users to find what they're looking for.

### CODE REVIEW

File: _layouts/base.html [hunk 1]
```diff
@@ -113,13 +115,31 @@ <h1><a href="{{ site.baseurl }}/">{{ site.title }}</a></h1>
...
+      function loadSearch(lang, site) {
+        docsearch({
+          apiKey:'a57ef92bf2adfae863a201ee43d6b5a1',
+          indexName: 'brew_all',
+          inputSelector: '#search-bar',
+          algoliaOptions: { 'facetFilters': [
+            'lang: ' + lang,
+            'site: ' + site,
+          ] },
+          debug: true
```
Comment (MikeMcQuaid) [inline _layouts/base.html]: Probably don't want `debug: true` in the production version?

File: _layouts/base.html [hunk 2]
```diff
@@ -113,13 +115,31 @@ <h1><a href="{{ site.baseurl }}/">{{ site.title }}</a></h1>
...
     </script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/4.2.0/anchor.min.js" 
             integrity="sha256-m1eTvwEHwmsw4+XKF7BshClVJEPwTVycveNl0CS0Mkk=" 
-            crossorigin="anonymous"></script>
-    <script type="text/javascript">
-      anchors.options.placement = 'left';
-      anchors.add('#page > h2, #page > h3, #page > h4, #page > h5, #page > h6');
-    </script>
+            crossorigin="anonymous"
+            onload="loadAnchors()"></script>
```
Comment (MikeMcQuaid) [inline _layouts/base.html]: I'm confused about how this change is related (if it is)? If not: could it go in another PR?

File: assets/css/styles.scss [hunk 3]
```diff
@@ -439,7 +439,7 @@ span .amp {
   border-radius: 5px;
 }
 
-#newsletter-email {
+#newsletter-email, search-bar {
```
Comment (MikeMcQuaid) [inline assets/css/styles.scss]: Needs a `#search-bar` for an ID.

**Comment by MikeMcQuaid** [comment]
@SMillerDev Any news on this?

**Comment by SMillerDev** [comment]
Great minds think alike, I was just doing some fixes here.

**Comment by SMillerDev** [comment]
@MikeMcQuaid any opinions?

**Comment by MikeMcQuaid** [comment]
@SMillerDev I'm not notified on force-pushes, FYI. Looking now.

**Comment by MikeMcQuaid** [comment]
Need to handle the failing test. Will checkout the branch and take a look locally.

**Comment by MikeMcQuaid** [comment]
Positioning seems weird. I can help with the CSS if desired: screenshot 2019-02-26 at 11 22 19

Getting JS errors. Do you see these locally? screenshot 2019-02-26 at 11 22 16

**Comment by SMillerDev** [comment]
Yeah, I wasn't really sure where to position it. Opinions on that are appreciated

**Comment by asantarissy** [comment]
> Yeah, I wasn't really sure where to position it. Opinions on that are appreciated

@SMillerDev how about something like this, and maybe add a search icon next to it:

screen shot 2019-02-26 at 1 38 37 pm

**Comment by SMillerDev** [comment]
screenshot 2019-02-26 at 15 12 29
This does look better. @MikeMcQuaid agree with the search in the header?

**Comment by MikeMcQuaid** [comment]
> agree with the search in the header?

yes 👍

**Comment by SMillerDev** [comment]
@MikeMcQuaid any more comments/requests?

**Comment by MikeMcQuaid** [comment]
Polished off the final bits! Thanks for this @SMillerDev 😍

File: _layouts/base.html [hunk 4]
```diff
@@ -129,7 +129,7 @@ <h1><a href="{{ site.baseurl }}/">{{ site.title }}</a></h1>
           apiKey:'a57ef92bf2adfae863a201ee43d6b5a1',
           indexName: 'brew_all',
           inputSelector: '#search-bar',
-          algoliaOptions: { 'facetFilters': [
+          algoliaOptions: { facetFilters': [
```
Comment (michaelblyons) [inline _layouts/base.html]: Hanging single quote?

Comment (MikeMcQuaid) [reply _layouts/base.html]: https://github.com/Homebrew/brew.sh/pull/334 thanks
